src/content/docs/apm/agents/go-agent/configuration/go-agent-code-level-metrics-config.mdx

```diff
@@ -0,0 +1,286 @@
+responsible for the actions being instrumented by those traces. This data includes:
+ * Source file name
+ * Source file line number
+ * Function name
+ * Namespace
+
+<Callout variant="important">
+    This feature is available for transactions as of Go agent version 3.18.0, and is turned off by default. To enable it, you must
+    add `newrelic.ConfigCodeLevelMetricsEnabled(true)` to your application's configuration as explained below. Future releases may
+    change this to be on by default and may expand the feature to be included with other trace types other than transactions.
+</Callout>
+
+## Configuration methods [#config]
+
+When calling `newrelic.NewApplication` to configure the Go agent in your application, you can enable code-level metrics by
+including the `newrelic.ConfigCodeLevelMetricsEnabled` option, like so:
+
+```go
+app, err := newrelic.NewApplication(
+   newrelic.ConfigAppName("Your Application Name"),
+   newrelic.ConfigLicense(os.Getenv("NEW_RELIC_LICENSE_KEY")),
+   newrelic.ConfigCodeLevelMetricsEnabled(true),
+)
+```
+
+With that simple step, the Go agent will add the source code context information in the following agent attributes on transactions:
+<table>
+  <thead>
+    <tr>
+    <th style={{ width: "200px" }}>
+      Name
+    </th>
+    <th>
+      Description
+    </th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+    <td>
+      `code.function`
+    </td>
+    <td>
+      The name of the function which started the transaction. Note that this may be automatically generated by the Go compiler in some cases.
+    </td>
+    </tr>
+    <tr>
+    <td>
+      `code.namespace`
+    </td>
+    <td>
+      The package/namespace where the function is located. The combination of `code.namespace` and `code.function`, joined with a delimiter such as
+      a dot (`.`), is expected to uniquely identify the function.
+    </td>
+    </tr>
+    <tr>
+    <td>
+      `code.filepath`
+    </td>
+    <td>
+      The pathname of the source file containing the function being reported. Normally this is the full, absolute pathname; see below for
+      a configuration option which allows you to change this to a relative path.
+    </td>
+    </tr>
+    <tr>
+    <td>
+      `code.lineno`
+    </td>
+    <td>
+      The line number within `code.filepath` where the transaction was started. See [the instrumentation guide](/docs/apm/agents/go-agent/instrumentation/go-agent-code-level-metrics-instrument) for ways to control this.
+    </td>
+    </tr>
+  </tbody>
+</table>
+
+There are additional configuration options you may wish to use to further refine the collection of code-level metrics in your application.
+
+<CollapserGroup>
+  <Collapser
+    id="go-clm-config-scope"
+    title={<>Restrict CLM scope</>}
+  >
+  By default, with code-level metrics enabled, the agent will calculate and store source code context information for all types of traces
```

Comment:
The repetition of "By default" kept snagging me. I'd be curious for other tech writers to weigh in but while I valued the clarity of the repetition, it also made me feel like I was repeating the same steps. Maybe you could solve this with an introductory statement that says something like, "This is configured with default settings. If you'd like to configure it according to your desired specifications, here are some tips.." (obviously don't use that language but you get the idea!)

Reply:
@homelessbirds we decided to remove this section since the default is to collect "all" types and there is currently only one type. So, we'll add this back in later when more types become available so users can control their types.